src/app/src/test/kotlin/org/vaccineimpact/api/tests/controllers/UserControllerTests.kt

```diff
@@ -0,0 +1,57 @@
+package org.vaccineimpact.api.tests.controllers
+
+import com.nhaarman.mockito_kotlin.*
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import org.vaccineimpact.api.app.ActionContext
+import org.vaccineimpact.api.app.controllers.ControllerContext
+import org.vaccineimpact.api.app.controllers.UserController
+import org.vaccineimpact.api.app.errors.UnknownObjectError
+import org.vaccineimpact.api.app.repositories.UserRepository
+import org.vaccineimpact.api.models.*
+import org.vaccineimpact.api.models.permissions.RoleAssignment
+
+class UserControllerTests : ControllerTests<UserController>()
+{
+    override fun makeController(controllerContext: ControllerContext)
+            = UserController(controllerContext)
+
+    @Test
+    fun `getUserByUsername returns user`()
+    {
+        val userName = "test"
+        val user = User("test", "test name", "test@test.com", null, listOf(RoleAssignment("member", "", "")))
+
+        val controllerContext = mockControllerContext(mock<UserRepository> {
+            on { this.getUserByUsername(userName) } doReturn user
+        })
+        val context = mock<ActionContext> {
+            on { hasPermission(any()) } doReturn true
+            on { params(":username") } doReturn userName
+        }
+
+        val controller = UserController(controllerContext)
+        assertThat(controller.getUser(context)).isEqualToComparingFieldByField(user)
+    }
+
+    @Test(expected = UnknownObjectError::class)
```

Comment:
I've been tending to use `assertThatThrownBy { ... }.inInstanceOf(UnknownObjectError::class.java)` rather than `expected` attribute. My reason for preferring this is that then you are asserting that just that line (or block of code) under test throws the exception. Otherwise if it were thrown by some part of the test setup (unlikely in this case I know) that would also count as a passing test.